src/core/TimeDeltaUI.h

```diff
@@ -0,0 +1,52 @@
+/*
+ *  Copyright (C) 2022 KeePassXC Team <team@keepassxc.org>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef KEEPASSXC_TIMEDELTAUI_H
+#define KEEPASSXC_TIMEDELTAUI_H
+
+#include <QObject>
+#include <map>
+#include "TimeDelta.h"
+
+class QSpinBox;
+class QComboBox;
+
+class TimeDeltaUI : public QObject
```

Comment:
This class is poorly named, it is a QObject not a UI element (typically QWidget). This would be better placed as a standard time picker type widget that could also be used in the expiration UI, for example. As it is, this is a helper class that converts time units and has no need to know about UI elements like spinboxes and comboboxes. The parent widget to those elements should handle the communication.